event.go

```diff
@@ -50,6 +50,14 @@ func newEventMessage(i interface{}) *EventMessage {
 	return &EventMessage{i: i}
 }
 
+// ToString format message to string
+func (p *EventMessage) ToString() (string, error) {
```

Comment:
I'd rather it implements the [Stringer](https://golang.org/pkg/fmt/#Stringer) interface directly. Instead of returning an error when it's not a `[]byte`, you can return `fmt.Sprintf("%#v", p.i)`